src/sonic-yang-mgmt/tests/yang-model-tests/yangModelTesting.py

```diff
@@ -0,0 +1,426 @@
+            self.logStartTest(tStr)
+            jInput = self.readJsonInput(func)
+            # load the data, expect a exception with must condition failure
+            s = self.loadConfigData(jInput)
+            if ("When condition" in s and "not satisfied" in s and "IP_TYPE" in s):
+                log.info(tStr + " Passed\n")
+                return PASS
+        except Exception as e:
+            printExceptionDetails()
+        log.info(tStr + " Failed\n")
+        return FAIL
+
+    """
+        Configure l4_src_port_range as 99999-99999 in ACL_RULE.
+    """
+    def yangTest12(self, func):
+        try:
+            tStr = " Configure l4_src_port_range as 99999-99999 in ACL_RULE."
+            self.logStartTest(tStr)
+            jInput = self.readJsonInput(func)
+            # load the data, expect a exception with must condition failure
+            s = self.loadConfigData(jInput)
+            if ("does not satisfy" in s and "pattern" in s):
+                log.info(tStr + " Passed\n")
+                return PASS
+        except Exception as e:
+            printExceptionDetails()
+        log.info(tStr + " Failed\n")
+        return FAIL
+
+    """
+        Configure IP_TYPE as ARP and ICMPV6_CODE in ACL_RULE.
+    """
+    def yangTest13(self, func):
+        try:
+            tStr = " Configure IP_TYPE as ARP and ICMPV6_CODE in ACL_RULE."
+            self.logStartTest(tStr)
+            jInput = self.readJsonInput(func)
+            # load the data, expect a exception with must condition failure
+            s = self.loadConfigData(jInput)
+            if ("When condition" in s and "not satisfied" in s and "IP_TYPE" in s):
+                log.info(tStr + " Passed\n")
+                return PASS
+        except Exception as e:
+            printExceptionDetails()
+        log.info(tStr + " Failed\n")
+        return FAIL
+
+
+    """
+        Configure INNER_ETHER_TYPE as 0x080C in ACL_RULE.
+    """
+    def yangTest14(self, func):
+        try:
+            tStr = " Configure INNER_ETHER_TYPE as 0x080C in ACL_RULE."
+            self.logStartTest(tStr)
+            jInput = self.readJsonInput(func)
+            # load the data, expect a exception with must condition failure
+            s = self.loadConfigData(jInput)
+            if ("does not satisfy" in s and "pattern" in s):
+                log.info(tStr + " Passed\n")
+                return PASS
+        except Exception as e:
+            printExceptionDetails()
+        log.info(tStr + " Failed\n")
+        return FAIL
+
+# End of Class
+
+"""
+    Start Here
+"""
+def main():
+    parser = argparse.ArgumentParser(description='Script to run YANG model tests',
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog="""
+Usage:
+python yangModelTesting.py -h
+""")
+    parser.add_argument('-t', '--tests', type=str, \
+    help='Tests to run, Format 1,2,3', required=True)
```

Comment:
Let's make it optional, if nothing specified, we should run all test cases.
And the format is better to be the function names themselves.

Reply:
Resolved. 

Run Selected test Cases:
```
admin@lnos-x1-a-asw01:~/praveen/sonic-yang-mgmt/tests/yang-model-tests$ python yangModelTesting.py  -t 'ACL_TABLE_UNDEFINED_TABLE_TYPE,ACL_RULE_ARP_TYPE_DST_IPV6_MISMATCH, ACL_RULE_IP_TYPE_SRC_IPV6_MISMATCH' -f yangTest.json -y ../../yang-models/
```
Run all tests:
```
admin@lnos-x1-a-asw01:~/praveen/sonic-yang-mgmt/tests/yang-model-tests$ python yangModelTesting.py -f yangTest.json -y ../../yang-models/
```